browser/utils/android/java/src/org/chromium/brave/browser/utils/DrawableUtils.java

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.brave.browser.utils;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapShader;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Shader;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+
+import androidx.core.content.ContextCompat;
+
+public class DrawableUtils {
```

Comment:
What is the point of this class? It has one public method called in one place.

Reply:
@bridiver i created this class mainly to add any drawable related functions in one place. we have several functions to perform modifications in drawable in main directory which we can migrate to this class in future.

Reply:
having a directory called `browser/utils` is already a big red flag for me. We shouldn't just be dumping random helpers into a directory like this. Do we really need something wrapped around `Drawable`? You shouldn't have a class that is a collection of "utility" methods used in only one place. If this is common code used in different places, you should create a separate PR to refactor it and remove the duplication. Also if there is common code to be shared, consider a class that encapsulates some behavior instead of a generic "util" class. I believe we have unit tests for android java code as well? So these kinds of things should have unit tests.

Reply:
if anything here I would encapsulate this code in some kind of CustomAppIcon class that also encapsulates the logic in updateCustomAppIcon

Reply:
@bridiver chromium has the same util package here : https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/util/android/java/src/org/chromium/chrome/browser/util/

and the use case is also the same as if you check : https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/util/android/java/src/org/chromium/chrome/browser/util/KeyNavigationUtil.java, they have mentioned static functions for key events.

Reply:
That code is very old, I would not use it as an example, but the static function is not the problem, the problem is that the code is only used in one place. Although I think that directory is a bad call in upstrea, what they have in there are much more generic utilities, not code used in a single place. Also despite having many places that use Drawable, they never felt the need to create some higher level abstraction on top of it so why do we need it?

Reply:
the latest class i see is from 2023. and the last addition in the class is in 2024. 
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/util/android/java/src/org/chromium/chrome/browser/util/BrowserUiUtils.java

Reply:
@bridiver we have a design system in place right now which defines common UI components across the app with uniform theme and styling. and one example of it is `Confirmation dialog`. i am working on a separate feature for custom search engines which also requires the same component. Where should we keep common components  to avoid duplication?

Reply:
also to be clear since I've seen this before in brave android code, utils are not a place to group a bunch of semi-related functions that are only used in one place, they are for shared code used in many places and ideally encapsulated in a class with state vs static utility functions. They should also be modularized to reduce dependencies and not arbitrarily grouped together.

Reply:
@bridiver basically i have added the changes in utils which i know for sure, we are going to use in future as per the design system.